tag.sh

```diff
@@ -1,14 +1,50 @@
 #!/bin/bash
-# TODO: don't run if current HEAD is already tagged (ensure this is idempotent)
-# take current HEAD with new version
+set -eEu
```

Comment:
I didn't even know there were this many options! ;)

Reply:
things one learns with shellcheck and style guides :-)